pkg/accounts/sessions.go

```diff
@@ -37,43 +38,43 @@ var sessions map[string]*SessionInformation
 func init() {
 	sessions = make(map[string]*SessionInformation)
 }
+var (
+	ErrInvalidCredentials = errors.New("invalid user credentials")
+	ErrGenerateSessionId  = errors.New("session id creation failure")
+)
 
 // CreateSession: create a new session and return it's information
-func CreateSession(ctx context.Context, username, password string) *SessionInformation {
+func CreateSession(ctx context.Context, username, password string) (*SessionInformation, error) {
+	logger := klog.FromContext(ctx)
 
 	var session *SessionInformation
 
 	// Validate user credentials
 	valid, err := AccountsHandler().ValidAccount(username, password)
-	if valid && err == nil {
-		session = CreateSessionToken(ctx, username)
+	if !valid || err != nil {
+		logger.V(1).Error(err, "failure: create session", "username", username)
+		return nil, ErrInvalidCredentials
 	}
 
-	return session
-}
-
-// CreateSessionToken: create a token and add it to the map, return a new session id
-func CreateSessionToken(ctx context.Context, user string) *SessionInformation {
-	logger := klog.FromContext(ctx)
-
 	// Example uuid: ee0328d9-258a-4e81-976e-b75aa4a2d8f5
 	token := uuid.New().String()
 	token = strings.ReplaceAll(token, "-", "")
 
-	// Determine the next session id to use
-	activeSessionId += 1
-	if activeSessionId > maxSessionId {
```

Comment:
I suggest to remove the const `maxSessionId` since we don't need it anymore

Reply:
Yep, meant to do that and forgot, thanks. Done.